include/xsparse/levels/hashed.hpp

```diff
@@ -36,8 +36,47 @@ namespace xsparse
                 typename BaseTraits::I const m_i;
 
             public:
-                using iterator_type = std::unordered_map<typename BaseTraits::IK,
-                                                         typename BaseTraits::PK>::const_iterator;
+                class iterator;
+                using value_type =
+                    typename std::pair<typename BaseTraits::IK, typename BaseTraits::PK>;
+                using difference_type = typename std::make_signed_t<typename BaseTraits::PK>;
+                using pointer =
+                    typename std::pair<typename BaseTraits::IK, typename BaseTraits::PK>*;
+                using reference = std::pair<typename BaseTraits::IK, typename BaseTraits::PK>;
+                using iterator_type = iterator;
+
+                class iterator : public xtl::xbidirectional_iterator_base2<iteration_helper>
```

Comment:
@hameerabbasi this is the iterator class and for defining ++, -- and == operators, I'll need a reference to an iterator of `iterator_type`. Should I wrap the `iterator` class inside another class to get this reference through the iterator of the vector or how should I go about doing it